src/test/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -0,0 +1,164 @@
+package org.opentripplanner.middleware.trip_monitor.jobs;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.OtpMiddlewareTest;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.models.OtpUser;
+import org.opentripplanner.middleware.models.TripMonitorNotification;
+import org.opentripplanner.middleware.otp.OtpDispatcher;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.LocalizedAlert;
+import org.opentripplanner.middleware.otp.response.Response;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.utils.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.opentripplanner.middleware.TestUtils.getBooleanEnvVar;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.createMonitoredTrip;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.createUser;
+import static org.opentripplanner.middleware.trip_monitor.jobs.CheckMonitoredTrip.shouldSkipMonitoredTripCheck;
+
+/**
+ * This class contains tests for the {@link CheckMonitoredTrip} job.
+ */
+public class CheckMonitoredTripTest extends OtpMiddlewareTest {
+    private static final Logger LOG = LoggerFactory.getLogger(CheckMonitoredTripTest.class);
+    private static OtpUser user;
+    private static final String mockResponse = FileUtils.getFileContents(
+        "src/test/resources/org/opentripplanner/middleware/persistence/planResponse.json"
+    );
+    private static final OtpDispatcherResponse otpDispatcherResponse = new OtpDispatcherResponse(mockResponse);
+
+    @BeforeAll
+    public static void setup() {
+        user = createUser("user@example.com");
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        Persistence.otpUsers.removeById(user.id);
+        Persistence.monitoredTrips.removeFiltered(eq("userId", user.id));
+    }
+
+    /**
+     * To run this trip, change the env.yml config values for OTP_SERVER
+     * (and OTP_PLAN_ENDPOINT) to a valid OTP server.
```

Comment:
Do we also need to change the user's email in this class?

Reply:
If we want to have notifications sent out, then perhaps yea.